packages/browser/src/integrations/launchdarkly.ts

```diff
@@ -0,0 +1,68 @@
+import type { Client, Event, EventHint, IntegrationFn } from '@sentry/types';
+import type { LDContext, LDEvaluationDetail, LDInspectionFlagUsedHandler } from 'launchdarkly-js-client-sdk';
+
+import { insertToFlagBuffer } from '../utils/flags'
+import { defineIntegration, getCurrentScope } from '@sentry/core';
+
+/**
+ * Sentry integration for capturing feature flags from LaunchDarkly.
+ *
+ * See the [feature flag documentation](https://develop.sentry.dev/sdk/expected-features/#feature-flags) for more information.
+ *
+ * @example
+ * ```
+ * import * as Sentry from '@sentry/browser';
+ * import {launchDarklyIntegration, buildLaunchDarklyFlagUsedInspector} from '@sentry/browser';
+ * import * as LaunchDarkly from 'launchdarkly-js-client-sdk';
+ *
+ * Sentry.init(..., integrations: [launchDarklyIntegration()])
+ * const ldClient = LaunchDarkly.initialize(..., {inspectors: [buildLaunchDarklyFlagUsedHandler()]});
+ * ```
+ */
+export const launchDarklyIntegration = defineIntegration(() => {
+  return {
+    name: 'LaunchDarkly',
+
+    processEvent(event: Event, _hint: EventHint, _client: Client): Event {
+      const scope = getCurrentScope();
+      const flagContext = scope.getScopeData().contexts.flags;
+
+      if (event.contexts === undefined) {
+        event.contexts = {};
+      }
+      event.contexts.flags = flagContext;
+      return event;
+    },
+  };
+}) satisfies IntegrationFn;
+
+/**
+ * LaunchDarkly hook that listens for flag evaluations and updates the `flags`
+ * context in our Sentry scope. This needs to be registered as an
+ * 'inspector' in LaunchDarkly initialize() options, separately from
+ * `launchDarklyIntegration`. Both are needed to collect feature flags on error.
+ */
+export function buildLaunchDarklyFlagUsedHandler(): LDInspectionFlagUsedHandler {
+  return {
+    name: 'sentry-flag-auditor',
+    type: 'flag-used',
+
+    synchronous: true,
+
+    /**
+     * Handle a flag evaluation by storing its name and value on the current scope.
+     */
+    method: (flagKey: string, flagDetail: LDEvaluationDetail, _context: LDContext) => {
+      if (typeof flagDetail.value === 'boolean') {
+        const scopeContexts = getCurrentScope().getScopeData().contexts;
+        if (!scopeContexts.flags) {
+          scopeContexts.flags = { values: [] };
+        }
+        const flagBuffer = scopeContexts.flags.values;
+        insertToFlagBuffer(flagBuffer, flagKey, flagDetail.value);
+        // TODO: the flag buffer isn't thread-safe, yet this handler and the event processor could access it at the same time.
```

Comment:
JS is single threaded

Reply:
TIL! ChatGPT clarified the concept of run-to-completion, I think we should be set